_studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw_par.cpp

```diff
@@ -2002,6 +2017,14 @@ mfxStatus CheckAndFixCtrl(
         }
     }
 
+    mfxExtVP9Segmentation* seg_video = GetExtBuffer(video);
+    if (seg_video)
+    {
+        const mfxExtVP9Param& extParSeg = GetExtBufferRef(video);
+        mfxInfoMFX video_par = video.mfx;
+        MFX_CHECK_STS(CheckSegmentationParam(*seg_video, extParSeg.FrameWidth, extParSeg.FrameHeight, caps, video_par, &ctrl));
```

Comment:
Please do not put functions call inside macro. See what happens after this macro expansion:
```
#define MFX_CHECK_STS(sts)              MFX_CHECK(MFX_SUCCEEDED(sts), sts)
MFX_CHECK(MFX_SUCCEEDED(CheckSegmentationParam(*seg_video, extParSeg.FrameWidth, extParSeg.FrameHeight, caps, video_par, &ctrl)), CheckSegmentationParam(*seg_video, extParSeg.FrameWidth, extParSeg.FrameHeight, caps, video_par, &ctrl));
```

This will cause double function call in case of error. Please, just use temporary variable:
```
mfxStatus sts = CheckSegmentationParam(*seg_video, extParSeg.FrameWidth, extParSeg.FrameHeight, caps, video_par, &ctrl);
MFX_CHECK_STS(sts);
```